src/api/middleware/files.js

```diff
@@ -0,0 +1,110 @@
+const HTTPStatus = require("http-status-codes");
+const fs = require("fs");
+const path = require("path");
+const util = require("util");
+const { MulterError } = require("multer");
+const multerConfig = require("../../config/multer");
+const { ErrorTypes } = require("./errorHandler");
+const cloudinary = require("cloudinary").v2;
+const env = require("../../config/env");
+const ValidationReasons = require("./validators/validationReasons");
+
+
+const save_folder = path.join(__dirname, "../../../public/");
+if (!fs.existsSync(save_folder)) fs.mkdirSync(save_folder);
+
+const parseError = (message) => message.toLowerCase().replace(" ", "-");
+
+const single = (field_name) => (req, res, next) => {
+    const upload = multerConfig.single(field_name);
+    upload(req, res, (error) => {
+        if (error || !req.file) {
```

Comment:
Place the `if(!req.file)` check before the upload call to make it fail even before this.

Then, you will only need an if(error) here, no more nested ifs. Also, the message variable can be `                const message = error instanceof MulterError ?
                    parseError(error.message) : error.message;`

Reply:
`req.file` is created by the `upload` function; checking it beforehand is impossible.

Reply:
that's it